src/templates/BlogPostTemplate/blogPostTemplate.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import ButtonBlock from "../../components/blog/ButtonBlock";
+import ImageBlock from "../../components/blog/ImageBlock";
+import QuoteBlock from "../../components/blog/QuoteBlock";
+import StatsBlock from "../../components/blog/StatsBlock";
+import TextBlock from "../../components/blog/TextBlock";
+import LogoSeparator from "./LogoSeparator.svg";
+
+type BlogPostTemplateProps = {
+  topic?: string;
+  title?: string;
+  author?: string;
+  date?: string;
+  headerImgAlt?: string;
+  headerImgUrl?: string;
+  slices?: readonly any[];
+};
+
+/** The JSX template we render for each blog post. */
+const BlogPostTemplate = ({
+  topic,
+  title,
+  author,
+  date,
+  headerImgUrl,
+  headerImgAlt,
+  slices,
+}: BlogPostTemplateProps) => {
+  // returns either date or author, or if both are given "date - author"
+  const getDateAuthorString = (): string | null => {
+    if (!date && !author) {
+      return null;
+    }
+    if (date) {
+      const formattedDate = new Date(date).toLocaleDateString("en-US", {
+        year: "numeric",
+        month: "long",
+        day: "numeric",
+      });
+      return author ? `${formattedDate} - ${author}` : formattedDate;
+    }
+    return author || null;
```

Comment:
So I ended up implementing this myself in my local branch for styling the Blog Index page, and I ended up with this, much more concise version:

```js
const formattedDate = date && new Date(date).toLocaleDateString(...); // This now has type `string | null`
return [formattedDate, author].filter((x) => x).join(" - ");
```

The `.filter()` gets rid of any null or empty strings, and the only difference between this and your version or my older version is that this new one will return an empty string instead of null, but I think it shouldn't matter because empty strings also count as `false`.